db/write_batch.cc

```diff
@@ -1475,50 +1471,70 @@ class MemTableInserter : public WriteBatch::Handler {
         std::string merged_value;
 
         auto cf_handle = cf_mems_->GetColumnFamilyHandle();
-        Status s = Status::NotSupported();
+        Status get_status = Status::NotSupported();
         if (db_ != nullptr && recovering_log_number_ == 0) {
           if (cf_handle == nullptr) {
             cf_handle = db_->DefaultColumnFamily();
           }
-          s = db_->Get(ropts, cf_handle, key, &prev_value);
+          get_status = db_->Get(ropts, cf_handle, key, &prev_value);
+        }
+        // Intentionally overwrites the `NotFound` in `ret_status`.
+        if (!get_status.ok() && !get_status.IsNotFound() &&
+            !get_status.IsNotSupported()) {
```

Comment:
Good catch. I thought it was ok because it's the initial value for this `Status` variable and normal cases like recovery would reach here without modifying this `Status`. Upon further thought, it looks like recovery really should fail when it reaches here; otherwise, users would see an inconsistent outcome w.r.t. whether update callback happened depending on whether there was a crash-recovery. That looks like another potential surprise for users (some crashes cannot be recovered...).